test/hotkey/key_window/check_key_window.c

```diff
@@ -0,0 +1,145 @@
+#include <hotkey/key_window.h>
+
+#include <hotkey/constant.h>
+
+#include <check.h>
+
+#include <cc_bstring.h>
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define SUITE_NAME "key_window"
+#define DEBUG_LOG  SUITE_NAME ".log"
+
+#define TEST_KEY_WINDOW_SIZE 10
+
+/*
+ * utilities
+ */
+static void
+test_setup(void)
+{
+    key_window_setup(TEST_KEY_WINDOW_SIZE);
+}
+
+static void
+test_teardown(void)
+{
+    key_window_teardown();
+}
+
+static void
+test_reset(void)
+{
+    test_teardown();
+    test_setup();
+}
+
+/**************
+ * test cases *
+ **************/
+
+START_TEST(test_basic)
+{
+    char *key1str = "key1";
+    char buf[MAX_KEY_LEN];
+    uint32_t key_window_pop_len;
+    struct bstring key1;
+
+    key1.data = key1str;
+    key1.len = strlen(key1str);
+
+    test_reset();
+
+    ck_assert_int_eq(key_window_len(), 0);
+
+    key_window_push(&key1);
+    ck_assert_int_eq(key_window_len(), 1);
+    key_window_pop_len = key_window_pop(buf);
+
+    ck_assert_int_eq(key_window_len(), 0);
+    ck_assert_int_eq(key_window_pop_len, key1.len);
+    ck_assert(strncmp(key1.data, buf, key1.len) == 0);
+}
+END_TEST
+
+START_TEST(test_multiple)
+{
+    char *key1str = "key1", *key2str = "key22", *key3str = "key333";
```

Comment:
use macros to define key1/2/3, as we did in other unit tests? (see protocol/data/memcache)

Also, you can use str2bstr to convert a C string to a struct bstring. https://github.com/twitter/ccommon/blob/master/include/cc_bstring.h#L38